main.nf

```diff
@@ -884,6 +890,57 @@ process Resample_B0 {
     """
 }
 
+dwi_for_extract_sh_fitting_shell
+    .join(gradients_for_sh_fitting_shell)
+    .set{dwi_and_grad_for_extract_sh_fitting_shell}
+
+process Extract_SH_Fitting_Shell {
+    cpus 3
+
+    input:
+    set sid, file(dwi), file(bval), file(bvec)\
+        from dwi_and_grad_for_extract_sh_fitting_shell
+
+    output:
+    set sid, "${sid}__dwi_sh_fitting.nii.gz", "${sid}__bval_sh_fitting",
+        "${sid}__bvec_sh_fitting" into \
+        dwi_and_grad_for_sh_fitting
+
+    when:
+    params.sh_fitting
+
+    script:
+    """
+    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
+    export OMP_NUM_THREADS=1
+    export OPENBLAS_NUM_THREADS=1
+    scil_extract_dwi_shell.py $dwi \
+        $bval $bvec 0 $params.sh_fitting_shell ${sid}__dwi_sh_fitting.nii.gz \
```

Comment:
I think we must to use the same kind of thing that fodf_shells "0 1000".

Maybe we can raise an error when we select sh_fitting and don't set sh_fitting_shell. You have an example with fodf/dti_shells

Reply:
Tested it on my side and it works perfectly, I get all b0 + b1000 extracted. the script arg is set to nargs='+', so it automatically detects all b-vals. I thought the "quote" thing was specifically for nextflow because it couldn't handle multiple args

Reply:
Sorry for the delay, I agree with @GuillaumeTh . you should have a warning when we select sh_fitting and don't set sh_fitting_shell and you should use the same kind of input as fodf_shells: "0 1000". Otherwise almost ready to be merged